Userland/Libraries/LibWeb/Layout/BlockFormattingContext.h

```diff
@@ -113,16 +113,36 @@ class BlockFormattingContext : public FormattingContext {
 
     struct BlockMarginState {
         Vector<float> current_collapsible_margins;
+        Optional<Function<void(float)>> update_box_waiting_for_final_y_position_callback;
 
         void add_margin(float margin)
         {
             current_collapsible_margins.append(margin);
         }
 
+        void add_block_container_waiting_for_final_y_position(Function<void(float)> callback)
```

Comment:
The name of this API makes it sound like we're adding a `BlockContainer` to something when in reality we're registering a callback function. Let's find a name that better matches what it actually does :)

Reply:
yes, makes sense. I used `block_container_y_position_update_callback` instead but probably still not perfect :)